### ISSUE TITLE
isDeterministic variable not initialized in SFT.MkSFT

### ISSUE THREAD

**Issue by GeorgeArgyros** [lorisdanto/symbolicautomata#17]
The isDeterministic variable is not initialized in the SFT.MkSFT() method and therefore deterministic transducers are classified as non-deterministic (since the variable is initialized to false in the Automaton class). 

Test case attached. 
[Main.txt](https://github.com/lorisdanto/symbolicautomata/files/2161298/Main.txt)

**Comment by shxuy** [comment]
Thanks for your test case and I have fixed the bug in my own branch.
I added codes to check determinism of SFT in method MkSFT (github.com/shxuy/symbolicautomata/blob/12d966368ff2ac48493bf951f547b3fd467d3cd6/SVPAlib/src/transducers/sft/SFT.java#L112, line 112 - 135). 
I also added your code in Main.txt and other test cases created by myself to SFTUnitTest (github.com/shxuy/symbolicautomata/blob/12d966368ff2ac48493bf951f547b3fd467d3cd6/TestSVPA/src/test/SFT/SFTUnitTest.java#L497, line 497 - 608)
Because I use professor's boolean algebra during the check of determinism, we have to throw TimeoutException when creating any SFTs. For example, in line 46 of your Main.txt, we have to add "throws TimeoutException" after "getTestSFTCounterexample()" now.
